testing/src/scenario/state.py

```diff
@@ -1047,6 +1069,13 @@ def get_filesystem(self, ctx: Context) -> pathlib.Path:
         """
         return ctx._get_container_root(self.name)
 
+    def get_check_info(self, name: str) -> CheckInfo:
+        """Get the check info for a check by name."""
+        for check_info in self.check_infos:
+            if check_info.name == name:
```

Comment:
🤔 `check_infos` is a frozen set.
CheckInfo doesn't implement custom `__hash__` by name only.
So, there could be multiple infos with the same name and let's say different levels, statuses or ids.

Is it meaningful to return a "random" check info in such case?
Or are we relying on unique names by construction somehow?